### PR TITLE
[issue-344] Correctly use function_modifier attribute when comparing definition uniqueness

### PR DESCRIPTION
**Pull request by ppeble**
This addresses one of the two scenarios identified in https://github.com/holidays/holidays/issues/344, specifically `Two regions each contains a function with the same name and same logic but also have different function modifier values`. 

The root cause of this issue is that we were incorrectly ignoring the function_modifier when deciding whether a definition was already loaded, resulting in cases where definitions were incorrectly merged in internal storage. This lead to scenarios where a region loaded second would execute with logic related to previously loaded regions.

This does not address the other scenario in the issue. That will require a more extensive refactoring and will be split out into a separate PR.

### CODE REVIEW

**Comment by ppeble** [comment]
FYI: the tests will fail until https://github.com/holidays/definitions/pull/132 is merged to master and I update to grab those changes.

File: test/integration/test_multiple_regions_with_conflict.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require File.expand_path(File.dirname(__FILE__)) + '/../test_helper'
+
+# See https://github.com/holidays/holidays/issues/344 for more info on why
+# these tests exist.
+class MultipleRegionsWithConflictsTests < Test::Unit::TestCase
+
+  def test_corpus_christi_returns_correctly_for_co_even_if_br_is_loaded_first
```
Comment (ttwo32) [inline test/integration/test_multiple_regions_with_conflict.rb]: Wow. This test is a perfect!!

File: test/holidays/definition/repository/test_holidays_by_month.rb [hunk 2]
```diff
@@ -152,4 +152,124 @@ def test_find_by_month_raises_error_if_month_is_not_valid
...
+          :name=>"Test",
+          :regions=>[:test2]
+        }
+      ]
+    }
+
+    assert_equal(expected, @subject.all)
+  end
+
+  def test_add_is_successful_if_only_function_modifier_is_different
```
Comment (ttwo32) [inline test/holidays/definition/repository/test_holidays_by_month.rb]: This test is good. 👍

**Comment by ttwo32** [comment]
OK. I will merge my [PR](https://github.com/holidays/definitions/pull/132) from now. 😄

**Pull request merged**